k2/csrc/ragged.h

```diff
@@ -212,6 +216,59 @@ class RaggedShape {
   std::vector<RaggedShapeLayer> layers_;
 };
 
+
+template <typename T, int MAX_DIM>
+struct ArrayAccessor {
+  T data[MAX_DIM];
+  ArrayAccessor() { }
+
+  __host__ __device__ ArrayAccessor(const ArrayAccessor &other) {
+      for (int i = 0; i < MAX_DIM; i++)
+        data[i] = other.data[i];
+  }
+
+};
+
+
+// call this variable `xxx_row_splits_acc`
+template <int MAX_LAYERS>
+struct RowSplitsAccessor {
+  int32_t *ptrs[MAX_LAYERS];  // these are indexed by layer, from 0.
+
+  // row_splits_acc(1) == shape.RowSplits(1), for instance.
+  int32_t *operator () (int32_t layer) { return ptrs[layer - 1]; }
```

Comment:
Do we need to add
```cpp
K2_DCHECK_GE(layer, 1);
K2_DCHECK_LE(layer, MAX_LAYERS);
```

Also, I would propose
```cpp
int32_t *ptr[MAX_LAYERS] = {nullptr};
```

Reply:
This class is intended to be used mostly on GPU; I don't want checks like this as they're slow and bulk up the compiled code.
I don't think it's necessary to initialize unused elements; it'll likely be a segfault if one is used regardless of the actual value.

Reply:
actually, initialization is convenient for debugging, will add = {nullptr}